gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/structure/io/gryo/GryoSerializersV3d0.java

```diff
@@ -427,7 +428,12 @@ else if (1 == arguments)
             output.writeString(object.getName());
             output.writeDouble(object.getDuration(TimeUnit.NANOSECONDS) / 1000000d);
             kryo.writeObject(output, object.getCounts());
-            kryo.writeObject(output, object.getAnnotations());
+
+            // annotations is a synchronized LinkedHashMap - get rid of the "synch" for serialization as gryo
+            // doesn't know how to deserialize that well and LinkedHashMap should work with 3.3.x and previous
+            final Map<String, Object> annotations = new LinkedHashMap<>();
+            object.getAnnotations().forEach(annotations::put);
+            kryo.writeObject(output, annotations);
 
             // kryo might have a problem with LinkedHashMap value collections. can't recreate it independently but
```

Comment:
Do we not need to worry about this comment?

Reply:
i wonder what that was in reference to.................we serder `LinkedHashMap` all over the place

Reply:
@robertdale i thought about it some more. the problem (whatever it was) wasn't related to `LinkedHashMap` itself which is what i'm serializing in the new code. the comment is in relation to the collection returned from `LinkedHashMap.values()` which is why I converted that collection to an `ArrayList`. better explanation?